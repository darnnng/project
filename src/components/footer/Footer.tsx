import React from 'react';
import { Link } from 'react-router-dom';
import { RoutePath } from '@constants/routes';
import styles from './Footer.module.scss';

export const Footer = () => {
  return (
    <footer className={styles.footer}>
      <div className={styles.footerUp}>
        <div>
          <p className={styles.footerBigTitle}>ABOUT US</p>
          <div className={styles.footerLinks}>
            <p>
              <Link className={styles.footerLinkText} to={`/${RoutePath.CATALOG}`}>
                Our stores
              </Link>
            </p>
            <p>
              <Link className={styles.footerLinkText} to={`/${RoutePath.CATALOG}`}>
                Catalog
              </Link>
            </p>
          </div>
        </div>
        <div className={styles.socialMedia}>
          <p className={styles.footerBigTitle}>CONTACTS</p>
          <div className={styles.footerLinksSocial}>
            <Link to={`https://www.facebook.com/`}>
              <span className={styles.materialSymbolsOutlined}>groups</span>
            </Link>
            <Link to={`https://www.google.com/gmail/about/`}>
              <span className={styles.materialSymbolsOutlined}>mail</span>
            </Link>
          </div>
        </div>
      </div>
      <div className={styles.footerDown}>
        <span>Threads & Co., Inc. All rights reserved</span>
      </div>
    </footer>
  );
};
