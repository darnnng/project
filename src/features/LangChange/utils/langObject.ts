import { Languages } from '@shared/constants/languages';

export const languages = [
  { code: Languages.EN, name: Languages.EN.toUpperCase() },
  { code: Languages.RU, name: Languages.RU.toUpperCase() },
];
