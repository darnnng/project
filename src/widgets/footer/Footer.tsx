import React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { RoutePath } from '@src/shared/constants/routes';
import styles from './Footer.module.scss';

export const Footer = () => {
  const { t } = useTranslation();
  return (
    <footer className={styles.footer}>
      <div className={styles.footerUp}>
        <div className={styles.aboutSection}>
          <p className={styles.footerBigTitle}>{t('ABOUT US')}</p>
          <div className={styles.footerLinks}>
            <p>
              <Link className={styles.footerLinkText} to={`/${RoutePath.CATALOG}`}>
                {t('Our stores')}
              </Link>
            </p>
            <p>
              <Link className={styles.footerLinkText} to={`/${RoutePath.CATALOG}`}>
                {t('Catalog')}
              </Link>
            </p>
          </div>
        </div>
        <div className={styles.socialMedia}>
          <p className={styles.footerBigTitle}>{t('CONTACTS')}</p>
          <div className={styles.footerLinksSocial}>
            <Link to={`https://www.facebook.com/`}>
              <span className={styles.materialSymbolsOutlined}>groups</span>
            </Link>
            <Link to={`https://www.google.com/gmail/about/`}>
              <span className={styles.materialSymbolsOutlined}>mail</span>
            </Link>
          </div>
        </div>
      </div>
      <div className={styles.footerDown}>
        <span>Threads & Co., Inc. All rights reserved</span>
      </div>
    </footer>
  );
};
