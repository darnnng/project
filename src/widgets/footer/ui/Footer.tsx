import { Link } from 'react-router-dom';
import { Trans, useTranslation } from 'react-i18next';
import { linkData, linkDataSocial } from '../model/links';
import styles from './Footer.module.scss';

export const Footer = () => {
  const { t } = useTranslation('translation');
  return (
    <footer className={styles.footer}>
      <div className={styles.footerUp}>
        <div className={styles.aboutSection}>
          <p className={styles.footerBigTitle}>
            <Trans>{t('about')}</Trans>
          </p>
          <div className={styles.footerLinks}>
            {linkData.map((link, index) => (
              <p key={index}>
                <Link className={styles.footerLinkText} to={`/${link.path}`}>
                  {t(link.text)}
                </Link>
              </p>
            ))}
          </div>
        </div>
        <div className={styles.socialMedia}>
          <p className={styles.footerBigTitle}>{t('contacts')}</p>
          <div className={styles.footerLinksSocial}>
            {linkDataSocial.map((link, index) => (
              <Link key={index} to={link.url}>
                <span className={styles.materialSymbolsOutlined}>{link.text}</span>
              </Link>
            ))}
          </div>
        </div>
      </div>
      <div className={styles.footerDown}>
        <span>Threads & Co., Inc. All rights reserved</span>
      </div>
    </footer>
  );
};
