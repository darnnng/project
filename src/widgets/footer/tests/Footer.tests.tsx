import { screen } from '@testing-library/react';
import { customRender } from '@shared/lib/utils.tests';
import { Footer } from '../Footer';

describe('Footer test', () => {
  test('renders elements', () => {
    customRender(<Footer />);
    screen.debug();
    const divElement = screen.getByText(/Catalog/i);
    const linkElements = screen.getAllByRole('link');
    expect(divElement).toBeInTheDocument();
    expect(linkElements).toHaveLength(4);
  });
  test('footer snapshot matches', () => {
    const { container } = customRender(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
