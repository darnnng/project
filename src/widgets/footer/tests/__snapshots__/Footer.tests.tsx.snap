// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test footer snapshot matches 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="footerUp"
    >
      <div
        class="aboutSection"
      >
        <p
          class="footerBigTitle"
        >
          ABOUT US
        </p>
        <div
          class="footerLinks"
        >
          <p>
            <a
              class="footerLinkText"
              href="/maps"
            >
              Our stores
            </a>
          </p>
          <p>
            <a
              class="footerLinkText"
              href="/catalog"
            >
              Catalog
            </a>
          </p>
        </div>
      </div>
      <div
        class="socialMedia"
      >
        <p
          class="footerBigTitle"
        >
          CONTACTS
        </p>
        <div
          class="footerLinksSocial"
        >
          <a
            href="https://www.facebook.com/"
          >
            <span
              class="materialSymbolsOutlined"
            >
              groups
            </span>
          </a>
          <a
            href="https://www.google.com/gmail/about/"
          >
            <span
              class="materialSymbolsOutlined"
            >
              mail
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="footerDown"
    >
      <span>
        Threads & Co., Inc. All rights reserved
      </span>
    </div>
  </footer>
</div>
`;

exports[`Footer test renders elements 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="footerUp"
    >
      <div
        class="aboutSection"
      >
        <p
          class="footerBigTitle"
        >
          ABOUT US
        </p>
        <div
          class="footerLinks"
        >
          <p>
            <a
              class="footerLinkText"
              href="/maps"
            >
              Our stores
            </a>
          </p>
          <p>
            <a
              class="footerLinkText"
              href="/catalog"
            >
              Catalog
            </a>
          </p>
        </div>
      </div>
      <div
        class="socialMedia"
      >
        <p
          class="footerBigTitle"
        >
          CONTACTS
        </p>
        <div
          class="footerLinksSocial"
        >
          <a
            href="https://www.facebook.com/"
          >
            <span
              class="materialSymbolsOutlined"
            >
              groups
            </span>
          </a>
          <a
            href="https://www.google.com/gmail/about/"
          >
            <span
              class="materialSymbolsOutlined"
            >
              mail
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="footerDown"
    >
      <span>
        Threads & Co., Inc. All rights reserved
      </span>
    </div>
  </footer>
</div>
`;
